include/realtime_tools/thread_priority.hpp

```diff
@@ -44,6 +46,13 @@ bool has_realtime_kernel();
  */
 bool configure_sched_fifo(int priority);
 
+/**
+ * Locks the memory pages of the calling thread to prevent page faults. By calling this method, the programs locks all pages mapped into the address space of the calling process and future mappings. This means that the kernel will not swap out the pages to disk i.e., the pages are guaranteed to stay in RAM until later unlocked - which is important for realtime applications.
+ * \param[out] message a message describing the result of the operation
+ * \returns true if memory locking succeeded, false otherwise
+*/
+bool lock_memory(std::string & message);
```

Comment:
Seems like a candidate to adopt `std::expected<std::monostate, std::string>`

Reply:
Oh, true. I was just musing anyway.

Reply:
No worries, it is interesting to learn this alternative for the future. Thank you